From 94f9d1eb18f60796b97de043d105b525612812a3 Mon Sep 17 00:00:00 2001
From: Abhijit Ayarekar <aayarekar@marvell.com>
Date: Fri, 2 Apr 2021 16:44:35 +0000
Subject: [PATCH 15/19] octeontx2-af: Update SDP AF driver to handle multiple
 SDP blocks

commit 9c3cc2afa031734daf28d811464b5ad448ac153c from
git@git.assembla.com:cavium/WindRiver.linux.git

Original commit f7621657cc954bcee00bebfa9334ae88ab6f5e8e

Some chips like 98xx has more than one SDP block. This patch handles
that case. Right now MAX_SDP is put to 2.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/29735
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>

Change-Id: I435a6ca474032361df3b09550f6280690bd87b42
Signed-off-by: Abhijit Ayarekar <aayarekar@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/49091
Reviewed-by: Radha Chintakuntla <radhac@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 .../ethernet/marvell/octeontx2/af/rvu_sdp.c   | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sdp.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sdp.c
index 8f41dafa6cfd..b4399d7d2045 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sdp.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sdp.c
@@ -14,12 +14,24 @@
 /* SDP PF device id */
 #define PCI_DEVID_OTX2_SDP_PF   0xA0F6
 
+/* Maximum SDP blocks in a chip */
+#define MAX_SDP		2
+
 /* SDP PF number */
-static int sdp_pf_num = -1;
+static int sdp_pf_num[MAX_SDP] = {-1, -1};
 
 bool is_sdp_pfvf(u16 pcifunc)
 {
-	if (rvu_get_pf(pcifunc) != sdp_pf_num)
+	u16 pf = rvu_get_pf(pcifunc);
+	u32 found = 0, i = 0;
+
+	while (i < MAX_SDP) {
+		if (pf == sdp_pf_num[i])
+			found = 1;
+		i++;
+	}
+
+	if (!found)
 		return false;
 
 	return true;
@@ -49,12 +61,13 @@ int rvu_sdp_init(struct rvu *rvu)
 			continue;
 
 		if (pdev->device == PCI_DEVID_OTX2_SDP_PF) {
-			sdp_pf_num = i;
+			sdp_pf_num[i] = pdev->bus->number - 1;
 			put_device(&pdev->dev);
 			break;
 		}
 
 		put_device(&pdev->dev);
+		i++;
 	}
 
 	return 0;
-- 
2.31.1

